src/DGtal/geometry/volumes/distance/LpMetric.h

```diff
@@ -0,0 +1,233 @@
+#include "DGtal/base/Common.h"
+#include "DGtal/kernel/CSpace.h"
+#include "DGtal/kernel/CEuclideanRing.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+/////////////////////////////////////////////////////////////////////////////
+// template class LpMetric
+/**
+ * Description of template class 'LpMetric' <p>
+ * \brief Aim: implements  l_p metrics.
+ *
+ * Given a parameter p, the class implements classical l_p
+ * metric as a model of CMetricSpace. Hence, given two points
+ * @f$ x=(x_0...x_{n-1})@f$, @f$ y=(y_0...y_{n-1})@f$, we define a metric as:
+ *
+ * @f$ distance(x,y)= \left(
+ * \sum_{i=0}^{n-1} |x_i-y_i|^p\right)^{1/p}@f$
+ *
+ * This class performs all computations on C++ double converting the digital
+ * points to Space::RealPoint
+ *
+ * The exponent @a p is specifed at the constructor.
+ *
+ * @tparam TSpace the model of CSpace on which the metric is
+ * defined.
+ */
+  template <typename TSpace>
+  class LpMetric
+  {
+    // ----------------------- Standard services ------------------------------
+  public:
+
+    ///Copy the space type
+    typedef TSpace Space;
+    BOOST_CONCEPT_ASSERT(( concepts::CSpace<TSpace> ));
+    
+    ///Type for points (RealPoint for this class)
+    typedef typename Space::RealPoint Point;
+    ///Type for distance values
+    typedef double Value;
+    ///Type for raw distance values
+    typedef double RawValue;
+    
+    /**
+     * Constructor.
+     *
+     * @param anExponent the exponent (@a p) of the lp metric.
+     */
+    LpMetric( const double anExponent): myExponent(anExponent)
+    {}
+
+
+    /**
+     * Destructor.
+     */
+    ~LpMetric()
+    {}
+
+    /**
+     * Copy constructor.
+     * @param other the object to clone.
+     */
+    LpMetric ( const LpMetric & other )
+    {
+      myExponent = other.myExponent;
+    }
+
+    /**
+     * Assignment.
+     * @param other the object to copy.
+     * @return a reference on 'this'.
+     */
+    LpMetric & operator= ( const LpMetric & other )
+    {
+      myExponent = other.myExponent;
+      return *this;
+    }
+
+    // ----------------------- Interface --------------------------------------
+  public:
+
+    // ----------------------- CMetric --------------------------------------
+    /**
+     * Compute the distance between @a aP and @a aQ.
+     *
+     * @param aP a first point.
+     * @param aQ a second point.
+     *
+     * @return the distance between aP and aQ.
+     */
+    Value operator()(const Point & aP, const Point &aQ) const
+    {
+      return std::pow( rawDistance(aP,aQ), 1.0/myExponent);
+    }
+
+    /**
+     * Compute the raw distance between @a aP and @a aQ.
+     *
+     * @param aP a first point.
+     * @param aQ a second point.
+     *
+     * @return the distance between aP and aQ.
+     */
+    RawValue rawDistance(const Point & aP, const Point &aQ) const
+    {
+      RawValue tmp=0.0;
+      for(auto i = 0; i < aP.size(); ++i)
```

Comment:
```suggestion
      for(typename Point::Dimension i = 0; i < aP.size(); ++i)
```
Otherwise, there is a warning because of a signed/unsigned comparison.

Reply:
👍  thanks